libnvidia-container-1.0.6.drv fails to build: missing rpcgen

**Describe the bug**
When I tried to run a system upgrade, it failed to build libnvidia-container. Trying to build this derivation separately also fails. I guess a missing dependency.
```
$ nix build /nix/store/gw0dz18pykg8zimmr3hs856szh1xqc92-libnvidia-container-1.0.6.drv
builder for '/nix/store/gw0dz18pykg8zimmr3hs856szh1xqc92-libnvidia-container-1.0.6.drv' failed with exit code 2; last 10 log lines:
  patching file src/nvc_info.c
  patching file src/nvc_ldcache.c
  configuring
  no configure script, doing nothing
  building
  build flags: SHELL=/nix/store/k8p54jg8ipvnfz435mayf5bnqhw4qqap-bash-4.4-p23/bin/bash WITH_LIBELF=yes prefix=\$\(out\)
  rm -f /build/source/src/driver_rpc.h
  cd /build/source/src/ && rpcgen -h -C -M -N -o driver_rpc.h /build/source/src/driver_rpc.x
  /nix/store/k8p54jg8ipvnfz435mayf5bnqhw4qqap-bash-4.4-p23/bin/bash: rpcgen: command not found
  make: *** [Makefile:174: /build/source/src/driver_rpc.h] Error 127
[0 built (1 failed)]
error: build of '/nix/store/gw0dz18pykg8zimmr3hs856szh1xqc92-libnvidia-container-1.0.6.drv' failed
```

**To Reproduce**
Steps to reproduce the behavior:
1. nix build /nix/store/gw0dz18pykg8zimmr3hs856szh1xqc92-libnvidia-container-1.0.6.drv

**Expected behavior**
libnvidia-container should build without errors.

**Additional context**
Add any other context about the problem here.

**Notify maintainers**
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

**Metadata**
these paths will be fetched (0.05 MiB download, 0.28 MiB unpacked):
  /nix/store/hfvjs6abgzadrv9f387v1ip5mr206jh9-bash-interactive-4.4-p23-dev
copying path '/nix/store/hfvjs6abgzadrv9f387v1ip5mr206jh9-bash-interactive-4.4-p23-dev' from 'https://cache.nixos.org'...
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.70, NixOS, 21.03pre246543.24c9b05ac53 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.7`
 - channels(root): `"nixos-21.03pre248170.007126eef72"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

I can also see the same problem here:

https://github.com/NixOS/nixpkgs/issues/89166

After looking around, I can't seem to find a proper rpcgen. I tried the following:

```nix
{ stdenv, lib, fetchFromGitHub, pkgconfig, libelf, libcap, libseccomp }:

with lib; let

  modp-ver = "396.51";

  nvidia-modprobe = fetchFromGitHub {
    owner = "NVIDIA";
    repo = "nvidia-modprobe";
    rev = modp-ver;
    sha256 = "1fw2qwc84k64agw6fx2v0mjf88aggph9c6qhs4cv7l3gmflv8qbk";
  };

in stdenv.mkDerivation rec {
  pname = "libnvidia-container";
  version = "1.0.6";

  src = fetchFromGitHub {
    owner = "NVIDIA";
    repo = "libnvidia-container";
    rev = "v${version}";
    sha256 = "1pnpc9knwh8d1zqb28zc3spkjc00w0z10vd3jna8ksvpl35jl7w3";
  };

  # locations of nvidia-driver libraries are not resolved via ldconfig which
  # doesn't get used on NixOS. Additional support binaries like nvidia-smi are
  # not resolved via the environment PATH but via the derivation output path.
  patches = [ ./libnvc-ldconfig-and-path-fixes.patch ];

  makeFlags = [
    "WITH_LIBELF=yes"
    "prefix=$(out)"
  ];

  postPatch = ''
    sed -i 's/^REVISION :=.*/REVISION = ${src.rev}/' mk/common.mk
    sed -i 's/^COMPILER :=.*/COMPILER = $(CC)/' mk/common.mk

    mkdir -p deps/src/nvidia-modprobe-${modp-ver}
    cp -r ${nvidia-modprobe}/* deps/src/nvidia-modprobe-${modp-ver}
    chmod -R u+w deps/src
    touch deps/src/nvidia-modprobe-${modp-ver}/.download_stamp
  '';

  nativeBuildInputs = [ pkgconfig ];

  buildInputs = [ libelf libcap libseccomp ];

  meta = {
    homepage = "https://github.com/NVIDIA/libnvidia-container";
    description = "NVIDIA container runtime library";
    license = licenses.bsd3;
    platforms = platforms.linux;
  };
}
```

that provides a rpcgen, but got errors

```
gcc -fPIC -std=gnu11 -O2 -g -fdata-sections -ffunction-sections -fstack-protector -fno-strict-aliasing -fvisibility=hidden -Wall -Wextra -Wcast-align -Wpointer-arith -Wmissing-prototypes -Wnonnull -Wwrite-strings -Wlogical-op -Wformat=2 -Wmissing-format-attribute -Winit-self -Wshadow -Wstrict-prototypes -Wunreachable-code -Wconversion -Wsign-conversion -Wno-unknown-warning-option -Wno-format-extra-args -Wno-gnu-alignof-expression  -DNV_LINUX -isystem /build/source/deps/nix/store/chz7gx3jbzqpsaj5hxyf2a7ws294q33i-libnvidia-container-1.0.6/include -include /build/source/src/build.h -DWITH_LIBELF -DWITH_SECCOMP -I/nix/store/mmcj5dvbjc6d9rac1hb0an658s3lbd0q-libseccomp-2.5.0-dev/include -D_GNU_SOURCE -D_FORTIFY_SOURCE=2  -DNDEBUG -MMD -MF /build/source/src/driver.d -c -o /build/source/src/driver.lo /build/source/src/driver.c
In file included from /build/source/src/driver.c:24:
/build/source/src/driver_rpc.h:9:10: fatal error: rpc/rpc.h: No such file or directory
    9 | #include <rpc/rpc.h>
      |          ^~~~~~~~~~~
compilation terminated.
make: *** [Makefile:177: /build/source/src/driver.lo] Error 1
```

--- a/pkgs/applications/virtualization/nvidia-docker/avoid-static-libtirpc-build.patch
+++ b/pkgs/applications/virtualization/nvidia-docker/avoid-static-libtirpc-build.patch
@@ -1,0 +1,21 @@
+diff --git a/Makefile b/Makefile
+index 0070ada..802cef0 100644
+--- a/Makefile
++++ b/Makefile
+@@ -202,7 +202,7 @@ $(BIN_NAME): $(BIN_OBJS)
+ ##### Public rules #####
+ 
+ all: CPPFLAGS += -DNDEBUG
+-all: shared static tools
++all: shared tools
+ 
+ # Run with ASAN_OPTIONS="protect_shadow_gap=0" to avoid CUDA OOM errors
+ debug: CFLAGS += -pedantic -fsanitize=undefined -fno-omit-frame-pointer -fno-common -fsanitize=address
+@@ -232,7 +232,6 @@ install: all
+ 	# Install header files
+ 	$(INSTALL) -m 644 $(LIB_INCS) $(DESTDIR)$(includedir)
+ 	# Install library files
+-	$(INSTALL) -m 644 $(LIB_STATIC) $(DESTDIR)$(libdir)
+ 	$(INSTALL) -m 755 $(LIB_SHARED) $(DESTDIR)$(libdir)
+ 	$(LN) -sf $(LIB_SONAME) $(DESTDIR)$(libdir)/$(LIB_SYMLINK)
+ 	$(LDCONFIG) -n $(DESTDIR)$(libdir)

--- a/pkgs/applications/virtualization/nvidia-docker/libnvc.nix
+++ b/pkgs/applications/virtualization/nvidia-docker/libnvc.nix
@@ -1,17 +1,23 @@
-{ stdenv, lib, fetchFromGitHub, pkgconfig, libelf, libcap, libseccomp }:
-
-with lib; let
-
+{ stdenv
+, lib
+, fetchFromGitHub
+, pkgconfig
+, libelf
+, libcap
+, libseccomp
+, rpcsvc-proto
+, libtirpc
+}:
+let
   modp-ver = "396.51";
-
   nvidia-modprobe = fetchFromGitHub {
     owner = "NVIDIA";
     repo = "nvidia-modprobe";
     rev = modp-ver;
     sha256 = "1fw2qwc84k64agw6fx2v0mjf88aggph9c6qhs4cv7l3gmflv8qbk";
   };
-
-in stdenv.mkDerivation rec {
+in
+stdenv.mkDerivation rec {
   pname = "libnvidia-container";
   version = "1.0.6";
 
@@ -22,19 +28,32 @@ in stdenv.mkDerivation rec {
     sha256 = "1pnpc9knwh8d1zqb28zc3spkjc00w0z10vd3jna8ksvpl35jl7w3";
   };
 
-  # locations of nvidia-driver libraries are not resolved via ldconfig which
-  # doesn't get used on NixOS. Additional support binaries like nvidia-smi are
-  # not resolved via the environment PATH but via the derivation output path.
-  patches = [ ./libnvc-ldconfig-and-path-fixes.patch ];
+  patches = [
+    # locations of nvidia-driver libraries are not resolved via ldconfig which
+    # doesn't get used on NixOS. Additional support binaries like nvidia-smi
+    # are not resolved via the environment PATH but via the derivation output
+    # path.
+    ./libnvc-ldconfig-and-path-fixes.patch
+
+    # the libnvidia-container Makefile wants to build and install static
+    # libtirpc libraries; this patch prevents that from happening
+    ./avoid-static-libtirpc-build.patch
+  ];
 
   makeFlags = [
     "WITH_LIBELF=yes"
     "prefix=$(out)"
+    # we can't use the WITH_TIRPC=yes flag that exists in the Makefile for the
+    # same reason we patch out the static library use of libtirpc so we set the
+    # define in CFLAGS
+    "CFLAGS=-DWITH_TIRPC"
   ];
 
   postPatch = ''
-    sed -i 's/^REVISION :=.*/REVISION = ${src.rev}/' mk/common.mk
-    sed -i 's/^COMPILER :=.*/COMPILER = $(CC)/' mk/common.mk
+    sed -i \
+      -e 's/^REVISION :=.*/REVISION = ${src.rev}/' \
+      -e 's/^COMPILER :=.*/COMPILER = $(CC)/' \
+      mk/common.mk
 
     mkdir -p deps/src/nvidia-modprobe-${modp-ver}
     cp -r ${nvidia-modprobe}/* deps/src/nvidia-modprobe-${modp-ver}
@@ -42,11 +61,14 @@ in stdenv.mkDerivation rec {
     touch deps/src/nvidia-modprobe-${modp-ver}/.download_stamp
   '';
 
-  nativeBuildInputs = [ pkgconfig ];
+  NIX_CFLAGS_COMPILE = [ "-I${libtirpc.dev}/include/tirpc" ];
+  NIX_LDFLAGS = [ "-L${libtirpc.dev}/lib" "-ltirpc" ];
 
-  buildInputs = [ libelf libcap libseccomp ];
+  nativeBuildInputs = [ pkgconfig rpcsvc-proto ];
 
-  meta = {
+  buildInputs = [ libelf libcap libseccomp libtirpc ];
+
+  meta = with lib; {
     homepage = "https://github.com/NVIDIA/libnvidia-container";
     description = "NVIDIA container runtime library";
     license = licenses.bsd3;

